Source/Dafny/Verifier/Translator.cs

```diff
@@ -11183,7 +11183,7 @@ public enum K { Free, Checked, Both }
         var module = f.EnclosingClass.EnclosingModuleDefinition;
         var functionHeight = module.CallGraph.GetSCCRepresentativePredecessorCount(f);
 
-        if (functionHeight < heightLimit && f.Body != null && RevealedInScope(f) && !(f.Body.Resolved is MatchExpr)) {
+        if (functionHeight < heightLimit && f.Body != null && RevealedInScope(f)) {
```

Comment:
It would be nice to have some of the larger cases in `TrSplitExpr`, such as this `expr is FunctionCallExpr && position` one, to be in a separate method. I know such extract method refactorings aren't immediately rewarding but making small refactorings in the code we touch should be the way forward.

If such as refactoring is in the first or last commit of a PR then I think it should have minimal impact on the reviewability.